Monika After Story/game/sprite-decoder.rpy

```diff
@@ -0,0 +1,346 @@
+                *fullcode
+            )
+
+        # if any sprite failed, or we didnt find a midfix
+        if not sprite_added:
+            return False, 0
+
+        # otherwise, we good to go
+        return True, 1 + increaseby
+
+    def __process_sweatdrop(spcode, index, export_dict, *prefixes):
+        """
+        Processes a sweatdrop off the given spcode at the given index
+
+        IN:
+            spcode the spcode to check
+            index the next index to check
+            export_dict - dict to add the sprite data to
+            prefixes letters to prefix the code with
+
+        OUT:
+            Tuple of the following format:
+                [0] - True if the sweatdrops were valid, False if not
+                [1] - the number of spots to increase the index by
+        """
+        # next letter must be the sweatdrop type
+        fullcode = list(prefixes)
+        fullcode.append(spcode[index])
+
+        sweatdrop = SWEAT_MAP.get("".join(fullcode), None)
+
+        if sweatdrop is None:
+            return False, 0
+
+        #All checks pass, we have sweat
+        export_dict["sweat"] = sweatdrop
+        return True, 1
+
+    def __process_tears(spcode, index, export_dict, *prefixes):
+        """
+        Processes a tear off the given spcode at the given index
+
+        IN:
+            spcode the spcode to check
+            index the next index to check
+            export_dict - dict to add the sprite data to
+            prefixes letters to prefix the code with
+
+        OUT:
+            Tuple of the following format:
+                [0] - True if the tears were valid, False if not
+                [1] - the number of spots to increase the index by
+        """
+        # next letter must be the tear type
+        fullcode = list(prefixes)
+        fullcode.append(spcode[index])
+
+        tears = TEAR_MAP.get("".join(fullcode), None)
+
+        if tears is None:
+            return False, 0
+
+        #Checks passed. Let's add this
+        export_dict["tears"] = tears
+        return True, 1
+
+    PROCESS_MAP = {
+        "b": __process_blush,
+        "s": __process_s,
+        "t": __process_tears,
+    }
+
+    SUB_PROCESS_MAP = {
+        "s": {
+            "d": __process_sweatdrop,
+        },
+    }
+
+    def parse_exp_to_kwargs(exp):
+        """
+        Converts exp codes to kwargs to pass into mas_drawmonika_rk
+
+        IN:
+            exp - spritecode to convert
+
+        OUT:
+            dict representing the exp as kwargs for mas_drawmonika_rk
+
+        ASSUMES:
+            exp is not in the staticsprite format (not exp_static)
+
+        RAISES:
+            - KeyError if pose, eyes, eyebrows, or mouth is invalid
+            - Exception if optional sprite is invalid
+        """
+        full_code = exp
+        kwargs = dict()
+
+        #First parse arms, taking care of leaning as well here
+        arms = ARM_MAP[exp[0]]
+        if arms == "def":
```

Comment:
"def" arms don't actually mean its always a "def" lean. It's intended to be possible to have different arms types for a lean type. that is why in sprite.py, `is_lean` just marks if a pose is leaning, and `position` can contain a tuple, which denotes its lean and arms type